plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClientModule.java

```diff
@@ -55,8 +62,14 @@ public static ConnectionFactory createConnectionFactory(BaseJdbcConfig config, C
 
         return new DriverConnectionFactory(
                 new Driver(),
-                config.getConnectionUrl(),
+                ensureMariaDbUrlCompatibility(config.getConnectionUrl()),
                 connectionProperties,
                 credentialProvider);
     }
+
+    @VisibleForTesting
+    static String ensureMariaDbUrlCompatibility(String connectionUrl)
```

Comment:
I'd rather prefer users are aware what driver is used since query parameters and configurations would vary. It's a breaking change but IMO a benefit over the longer run.

Reply:
Right - I'm just trying to make sure an upgrade doesn't "break" anything unnecessarily (i.e., users don't *have to* change the URL / worry about config changes during any upgrade, etc). 
But the docs have been updated to say they "may want" to change the URL.
It's a tricky call, I think a "compatibility layer" is a pragmatic compromise (less trouble for folks who don't care, and folks who do care can make the change if they so please, etc).

Reply:
@hashhar - I'm going to stick with this unless there's a compelling reason to break consumers upon upgrade.

Reply:
Every such "feature" is a maintainence cost and hides the fact that the MariaDB driver isn't being used and hence connection will not accept properties valid for MariaDB driver but not supported by MemSQL driver.
There will also be behaviour differences between drivers which will surprise people over time.

While such compatibility bridge looks useful in the short term I see more downsides in the long-term. (This is one reason why I don't find switching the driver to be "useful" at this moment anyway). If we do intend to switch the driver then I'd prefer to see breakage here rather than applying magic and hiding the detail from people who upgrade.

Reply:
@hashhar - understood.

However, SingleStore 1.0.1 is compatible with MariaDB 2.7.6 - so unless one of them diverges (and we uptake that change), transparently changing `mariadb` to `singlestore` to satisfy our internally updated driver *cannot* have any future consequences.

If a user attempts to use a feature in MariaDB that is not supported by SingleStore, that feature would have to have been introduced in MariaDB after 2.7.6 - and any reference in Trino Docs will show the MemSQL Plugin uses SingleStore driver (and a link to the properties supported by SingleStore).

I don't mind removing it from the PR, but I think there is an argument that this is the correct course of action - until we encounter something in a later SingleStore driver which breaks compatibility - in which case we could remove the `mariadb` handling, thus forcing users to change their configuration (and therefore become aware of the driver change - but only at a time when there is a breaking change - that time becomes the next release if we remove the "compatUrl" code - so I'm 50/50 whether it's better to force consumers to make a change we could handle transparently)).